src/agent/onefuzz/src/expand.rs

```diff
@@ -108,9 +110,29 @@ impl<'a> Expand<'a> {
             PlaceHolder::InputFileSha256.get_string(),
             ExpandedValue::Mapping(Box::new(Expand::input_file_sha256)),
         );
+        values.insert(
+            PlaceHolder::MachineId.get_string(),
+            ExpandedValue::Mapping(Box::new(Expand::machine_id)),
+        );
+
         Self { values }
     }
 
+    // by making this a Mapping rather than when creating an Expand instance,
+    // users only pay for fetching the machine_id when it's used in expansion
+    fn machine_id(&self, _format_str: &str) -> Option<ExpandedValue<'a>> {
+        let rt = tokio::runtime::Runtime::new().unwrap();
+        let result = rt.block_on(get_machine_id());
```

Comment:
I don't like creating a short-lived runtime here, but I understand why you did it.

But since this operation truly is fallible, why not return an error, instead of sinking errors into `None`?

Reply:
The rest of the Expand infrastructure doesn't support failable operations.  It would be pretty large change to propagate Result throughout all of the `Expand` and all of it's callers.

Reply:
Hmm, apparently we similarly use `ok()` for input file hashes in [this](https://github.com/microsoft/onefuzz/blob/main/src/agent/onefuzz/src/expand.rs#L114) private method.

The `evaluate_` methods are already fallible, since the template string may be malformed. Then the trick is just making sure that the callbacks in `Mapping` placeholders have the right return type. But since they're private, I don't think that requires any visible change for users of `Expand`.

Reply:
Vetted my thought here: #1217

If we choose to take that PR, then you can just use `?` as needed.